src/output/tree.rs

```diff
@@ -422,6 +423,14 @@ impl<'module, W: io::Write> Context<'module, W> {
                 self.write_atomic_format(format)
             }
 
+            Format::MapExpr(expr, format) => {
+                write!(&mut self.writer, "map (")?;
+                self.write_expr(expr)?;
+                write!(&mut self.writer, ")")?;
+                write!(&mut self.writer, " ")?;
+                self.write_atomic_format(format)
+            }
+
```

Comment:
We’ll want to push a name onto the context, as we do in `Context::write_record`. We should also push a value as well, but we don’t currently have a way of representing open values yet. It shouldn’t matter in the short term, but it would probably be a good idea to add a FIXME about that.

```suggestion
            Format::MapExpr(expr, format) => {
                let name = "x";
                write!(&mut self.writer, "map (fun {name} => ")?;
                self.names.push(name.to_string());
                // FIXME: push value
                self.write_expr(expr)?;
                self.names.pop();
                write!(&mut self.writer, ") ")?;
                self.write_atomic_format(format)
            }

```

Reply:
Thanks, I've made that change and it makes sense!